### PR TITLE
Create DockerBuild workflow in GitHub

### PR DESCRIPTION
**Pull request by romaninsh**
- moved container image building to codefresh
 - now uses docker layer caching, so most builds are fast
 - now has php trigger, so should auto-run when php releases new image
 - deduplicated code, now "Docker.template" is used for 7.2. 7.3 and 7.4 images
 - added support for xdebug
 - "candidate" mechanism and testing. Image will be pushed as "candidate-7.3" then if `test.php` passes, will be promoted to "7.3"
 - test.php can be extended to make sure image satisfies build requirements (although failing test on any version would prevent any candidates from being promoted)
 - added status badge and link to pipeline

### CODE REVIEW

File: .codefresh/deploy-build-image.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  - test
+  - push
+steps:
+  main_clone:
+    stage: prepare
+    type: git-clone
+    repo: atk4/image
+    revision: "${{CF_BRANCH}}"
+
+  fix_sha:
```
Comment (mvorisek) [inline .codefresh/deploy-build-image.yaml]: probably no `fix_sha`

File: Makefile [hunk 2]
```diff
@@ -2,6 +2,16 @@ SUBDIRS = $(wildcard [0-9].*)
...
+clean:
+	rm ?.?/Dockerfile
+
+build: clean all
+	(cd 7.2; docker build . -t atk4/image:7.2)
+	(cd 7.3; docker build . -t atk4/image:7.3)
+	(cd 8.0.0alpha1; docker build . -t atk4/image:7.3)
+
+test:
+	docker run -v "$PWD":/usr/src/app -it atk4/image:candidate-8.0 php /usr/src/app/test.php
```
Comment (mvorisek) [inline Makefile]: add EOL

File: Makefile [hunk 3]
```diff
@@ -2,6 +2,16 @@ SUBDIRS = $(wildcard [0-9].*)
...
+	@[ -f $@/Dockerfile ] || cat Dockerfile.template  | sed 's/^FROM.*/FROM php:$@-alpine/' > $@/Dockerfile | echo "Created $@/Dockerfile from Dockerfile.template"
 
 
+clean:
+	rm ?.?/Dockerfile
+
+build: clean all
+	(cd 7.2; docker build . -t atk4/image:7.2)
+	(cd 7.3; docker build . -t atk4/image:7.3)
+	(cd 8.0.0alpha1; docker build . -t atk4/image:7.3)
```
Comment (mvorisek) [inline Makefile]: prefer some loop, the 3rd one is even wrong I think (no alpha, why 7.3 tag twice, ..)

File: Dockerfile.template [hunk 4]
```diff
@@ -1,4 +1,5 @@
 FROM php:7.4-alpine
+# ^^ "make" will override verison
```
Comment (mvorisek) [inline Dockerfile.template]: version

File: 8.0.0beta3/Dockerfile [hunk 5]
```diff
@@ -6,12 +6,15 @@ RUN apk add icu-dev mysql-client postgresql-client postgresql-dev bash npm git j
...
-RUN touch /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
-
+RUN git clone https://github.com/xdebug/xdebug \
+    && cd xdebug \
+    # the last working commit, because the php-src is not up to date yet in this alpine
+    && phpize \
+    && ./configure --enable-xdebug-dev \
+    && make all \
+    && make install \
+    && echo "zend_extension=xdebug.so" >> /usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
```
Comment (mvorisek) [inline 8.0.0beta3/Dockerfile:17]: I think we should use `docker-php-ext-install` or simillar commands, see "install pdo_oci PHP extension"

File: .codefresh/deploy-build-image.yaml (hunk 1 above)
Comment (romaninsh) [reply .codefresh/deploy-build-image.yaml]: thx fixed.,

File: Makefile (hunk 3 above)
Comment (romaninsh) [reply Makefile]: fixed. Makefile syntax is pretty bad, no loop, at least i'm not familiar with loops in makefiles. Added comments.

File: 8.0.0beta3/Dockerfile (hunk 5 above)
Comment (romaninsh) [reply 8.0.0beta3/Dockerfile:17]: i tried and it didn't work for xdebug.

**Pull request merged**